plugins/destination/postgresql/client/migrate.go

```diff
@@ -167,6 +170,56 @@ func (c *Client) dropTable(ctx context.Context, tableName string) error {
 	return nil
 }
 
+func (c *Client) migrateToCQID(ctx context.Context, tableName string, _ schema.Column) (err error) {
+	// Steps:
+	// acquire connection
+	var conn *pgxpool.Conn
+	conn, err = c.conn.Acquire(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to acquire connection: %w", err)
+	}
+	defer conn.Release()
+	// start transaction
+	tx, err := conn.BeginTx(ctx, pgx.TxOptions{
+		IsoLevel: pgx.Serializable,
+	})
+	if err != nil {
+		return fmt.Errorf("failed to begin transaction: %w", err)
+	}
+
+	defer func() {
+		if err == nil {
+			err = tx.Commit(ctx)
+			if err != nil {
+				c.logger.Error().Err(err).Msg("failed to commit transaction")
+			}
+		}
+		if err != nil {
+			if rollbackErr := tx.Rollback(ctx); rollbackErr != nil {
+				c.logger.Error().Err(rollbackErr).Str("table", tableName).Msg("Failed to rollback transaction")
+			}
+		}
+	}()
+
+	sanitizedTableName := pgx.Identifier{tableName}.Sanitize()
+	sanitizedPKName := pgx.Identifier{getPKName(&schema.Table{Name: tableName})}.Sanitize()
+
+	// Drop existing primary key
+	_, err = tx.Exec(ctx, "alter table "+sanitizedTableName+" drop constraint "+sanitizedPKName)
+	if err != nil {
+		c.logger.Error().Err(err).Str("table", tableName).Msg("Failed to drop primary key")
+		return err
+	}
+
+	// Create new Primary Key with CQID
+	_, err = tx.Exec(ctx, "ALTER TABLE "+sanitizedTableName+" ADD CONSTRAINT "+sanitizedPKName+" Primary Key ("+pgx.Identifier{schema.CqIDColumn.Name}.Sanitize()+")")
```

Comment:
a nit: use the same casing across the queries added (L 208 uses lower-case, this one is mixed)